src/app/pages/ConnectorDetailsPage/ConfigurationStep.tsx

```diff
@@ -31,8 +31,9 @@ export const ConfigurationStep: FC<ConfigurationStepProps> = ({
 }) => {
   const { t } = useTranslation();
 
-  const formConfiguration = JSON.parse(JSON.stringify(configuration));
-  clearSecretEmptyValue(formConfiguration);
+  const formConfiguration = clearEmptyObjectValues(
+    JSON.parse(JSON.stringify(configuration))
```

Comment:
it will be better to move this to `clearEmptyObjectValues` function itself

Reply:
The `JSON.parse(JSON.stringify(configuration))` part?  I was going to ask why these have been necessary, I think these configuration objects should treated as immutables, and so these kinds of deep copies shouldn't be required.  I mean, stuff happens, but yeah, revisiting these patterns and why they're needed is worthwhile as they can hide some stuff.

Reply:
I had some doubts about considering immutables, maybe we can connect separately for this discussion.